### PR TITLE
[spirv] Add support for .SampleCmp() and .SampleCmpLevelZero()

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

File: tools/clang/test/CodeGenSPIRV/texture.sample-cmp.hlsl [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+// CHECK: [[v2fc:%\d+]] = OpConstantComposite %v2float %float_0_1 %float_0_2
+// CHECK: [[v3fc:%\d+]] = OpConstantComposite %v3float %float_0_1 %float_0_2 %float_0_3
+
+float4 main(int2 offset: A, float comparator: B) : SV_Target {
+// CHECK:              [[t1:%\d+]] = OpLoad %type_1d_image %t1
+// CHECK-NEXT:   [[gSampler:%\d+]] = OpLoad %type_sampler %gSampler
+// CHECK-NEXT: [[comparator:%\d+]] = OpLoad %float %comparator
+// CHECK-NEXT: [[sampledImg:%\d+]] = OpSampledImage %type_sampled_image [[t1]] [[gSampler]]
+// CHECK-NEXT:            {{%\d+}} = OpImageSampleDrefImplicitLod %float [[sampledImg]] %float_0_1 [[comparator]] ConstOffset %int_5
```
Comment (ehsannas) [inline tools/clang/test/CodeGenSPIRV/texture.sample-cmp.hlsl]: OpImageSampleDrefImplicitLod :
> Result Type must be a scalar of integer type or floating-point type. It must be the same as Sampled Type of the underlying OpTypeImage.

Maybe change one of these to return `int` type?

File: tools/clang/test/CodeGenSPIRV/texture.sample-cmp-level-zero.hlsl [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+// CHECK: [[v2fc:%\d+]] = OpConstantComposite %v2float %float_0_1 %float_0_2
+// CHECK: [[v3fc:%\d+]] = OpConstantComposite %v3float %float_0_1 %float_0_2 %float_0_3
+
+float4 main(int2 offset: A, float comparator: B) : SV_Target {
+// CHECK:              [[t1:%\d+]] = OpLoad %type_1d_image %t1
+// CHECK-NEXT:   [[gSampler:%\d+]] = OpLoad %type_sampler %gSampler
+// CHECK-NEXT: [[comparator:%\d+]] = OpLoad %float %comparator
+// CHECK-NEXT: [[sampledImg:%\d+]] = OpSampledImage %type_sampled_image [[t1]] [[gSampler]]
+// CHECK-NEXT:            {{%\d+}} = OpImageSampleDrefExplicitLod %float [[sampledImg]] %float_0_1 [[comparator]] Lod|ConstOffset %float_0 %int_5
```
Comment (ehsannas) [inline tools/clang/test/CodeGenSPIRV/texture.sample-cmp-level-zero.hlsl]: OpImageSampleDrefExplicitLod :
> Result Type must be a scalar of integer type or floating-point type. It must be the same as Sampled Type of the underlying OpTypeImage.

Maybe change one of these to return `int` type?

Comment (antiagainst) [reply tools/clang/test/CodeGenSPIRV/texture.sample-cmp-level-zero.hlsl]: We cannot sample from non-float texture types.

File: tools/clang/test/CodeGenSPIRV/texture.sample-cmp.hlsl (hunk 1 above)
Comment (antiagainst) [reply tools/clang/test/CodeGenSPIRV/texture.sample-cmp.hlsl]: Same here. Cannot sample from non-float textures.

Comment (ehsannas) [reply tools/clang/test/CodeGenSPIRV/texture.sample-cmp.hlsl]: Aah! I see :)

**Pull request merged**